coffee_maker_navigator_2/lib/di/di/dependency_container_manager.dart

```diff
@@ -0,0 +1,137 @@
+  ///
+  /// [subscriber]: An object that subscribes to the container of type [C]. The presence
+  /// of subscribers influences the creation and destruction of the container.
+  void subscribeToContainer<C>(Object subscriber) {
+    final currentSubscribers = _containerSubscribers[C] ??= {};
+    // Add the subscriber to the set of subscribers for the container type C if not already present.
+    if (currentSubscribers.add(subscriber)) {
+      _manageContainerLifecycle<C>();
+    }
+  }
+
+  /// Removes a subscriber for the container of type [C] and manages the container's lifecycle.
+  ///
+  /// This method unregisters a subscriber for the specified container type [C] and
+  /// manages the lifecycle of the container based on the remaining subscribers.
+  ///
+  /// [subscriber]: An object that unsubscribes from the container of type [C]. The absence
+  /// of subscribers can lead to the destruction of the container.
+  void unsubscribeFromContainer<C>(Object subscriber) {
+    final currentSubscribers = _containerSubscribers[C];
+    // Remove the subscriber from the set of subscribers for the container type C if present.
+    if (currentSubscribers != null && currentSubscribers.remove(subscriber)) {
+      _manageContainerLifecycle<C>();
+    }
+  }
+
+  /// Retrieves the container of type [C].
+  ///
+  /// This method returns the active container of the specified type [C]. If the container
+  /// does not exist, an error is thrown.
+  ///
+  /// Throws [StateError] if the container of type [C] does not exist.
+  ///
+  /// Returns an instance of the container of type [C].
+  C getContainer<C>() {
+    final container = _activeContainers[C];
+    if (container == null) {
+      throw StateError('Container of type $C does not exist.');
+    }
+    return container as C;
+  }
+
+  /// Handles the creation and destruction of containers based on the subscribers' state.
+  ///
+  /// This method manages the lifecycle of containers by creating a container when
+  /// there are active subscribers and disposing of the container when there are no
+  /// active subscribers.
+  void _manageContainerLifecycle<C>() {
+    final currentSubscribersForTypeC = _containerSubscribers[C];
+
+    if (currentSubscribersForTypeC == null ||
+        currentSubscribersForTypeC.isEmpty) {
+      // Dispose the container and remove it from the active list if there are no subscribers.
+      final currentContainer = _activeContainers[C];
+      if (currentContainer != null) {
+        currentContainer.dispose();
+        _activeContainers.remove(C);
+      }
+    } else if (!_activeContainers.containsKey(C)) {
+      // Create the container if it doesn't exist and there are subscribers.
+      final containerBuilder = _containerFactories[C];
+      if (containerBuilder != null) {
+        _activeContainers[C] = containerBuilder(_appLevelDependencies);
+      } else {
+        throw StateError('No container factory registered for type $C.');
```

Comment:
Also might be a good to provide some instruction how to manage it

Reply:
Gave hints here: aadaa25bbcf498b82a7888be911e49b4120319a5